content.rb

```diff
@@ -1,96 +1,71 @@
-require 'google/apis/analyticsreporting_v4'
+require "google/analytics/data/v1beta"
 require 'date'
 require 'json'
 
 class Content
-  attr_reader :service, :credentials
+  attr_reader :client
   def initialize
-    @service = Google::Apis::AnalyticsreportingV4::AnalyticsReportingService.new
-    @credentials = Google::Auth::ServiceAccountCredentials.make_creds(
-      scope: "https://www.googleapis.com/auth/analytics.readonly"
-    )
-    @service.authorization = @credentials
+    @client = ::Google::Analytics::Data::V1beta::AnalyticsData::Client.new
   end
 
   def most_popular_govuk_pages
-    format_response.to_json
+    rows = response_hash[:rows]
+    formatted = []
+    rows.each do |row|
+      row_data = {
+        page_views: row[:metric_values][0][:value],
+        page_path: row[:dimension_values][0][:value],
+        page_title: row[:dimension_values][1][:value]
```

Comment:
Maybe we could `dig` these, but I'm a bit concerned about the `[1]` - how sure are we this won't change? Not a problem, but I wonder what is in `[0]`?

Reply:
I changed it to:

```
row_data = {
          page_views: row.dig(:metric_values).first.dig(:value),
          page_path: row.dig(:dimension_values).first.dig(:value),
          page_title: row.dig(:dimension_values)[1].dig(:value)
        }
```

yeah the structure it returns is quite nested and horrible!

Reply:
Fantastic - thanks @Rosa-Fox 👍  Not much we can do about the API response, but does seem fragile